src/app/src/app.reducers.js

```diff
@@ -49,12 +51,20 @@ const appReducer = createReducer(
                     $set: true,
                 },
             }),
-        [setSensorData]: (state, payload) =>
+        [completePollingSensor]: (state, payload) =>
```

Comment:
Might be for future work but it seems like it'd make sense/be helpful to store the data and fetching states for each sensor separately in order to allow some to fail and others to succeed etc. Right now it looks like:

- the app sets up three regular polling actions, one for each sensor, which means three requests apiece every ten minutes or so
- each time one of the polling actions completes, the entire `sensorData` state gets updated regardless of which sensor the success response returned from

In #63, we're going to use `getState` inside one of these sensor data request actions, which will get a snapshot of the sensor data at the moment that particular request has started: https://github.com/azavea/ism-watershed-wellness-snapshot/pull/63/files#diff-2797588f438e153594e9908f359cc237R15

I'm worried that keeping all of these sensors together in the same piece of data will lead to unanticipated overwrites of data for other sensors, since each request will succeed or fail in its own time and it may be that some of them succeed and others fail etc. It's easy to reason about the order the Redux actions would run in when they are called, but harder to reason about the order in which the API responses will call them.

If we'd prefer not to split the sensor data into three separate pieces of the Redux store, it might be worthwhile to update the single sensor fetching action such that it makes all three API requests internally and then only does the `then` or `catch` blocks to dispatch actions on `Promise.all`. Doing that would also circumvent needing to set up three `setInterval` functions in the App component, since there could be just one function that was doing all the work.

Reply:
Thanks for the detailed comment. Good points. Leaning towards keeping the data fetching for each sensor separate since their availability is very different. Going to open an issue since this PR is blocking.

Reply:
Curious about the evolution of this comment. In #20 we moved away from `Promise.all`, which was the original implementation.

Reply:
re: `Promise.all`: I think keeping three separate sensor data/fetching objects in the Redux store seems preferable since it makes it easier to disentangle the fetching state of one sensor from another.

However, if it was desirable to keep the sensor data all as a single thing in the store -- and update it fully each time the `completeFetchSensorData` action is received -- it'd make future debugging easier to be able to know for sure when the full update would happen, which could be done via `Promise.all`.

Reply:
https://github.com/azavea/ism-watershed-wellness-snapshot/issues/64